### ISSUE TITLE
Unable to manage yubikey certificates when country code >2 characters

### ISSUE THREAD

**Issue by precurse** [Yubico/yubikey-manager-qt#165]
I created a CSR with my Yubikey with ECCP256. I signed the certificate, and was able to import the created certificate through the Yubikey Manager on Linux.

However, once it imported, I got kicked back to the main page with the following error: 

```
Unknown error: Country code name must be a 2 character country code
```

The issue is I can't get back in to removed the certificate at all, since it refuses to let me into the PIV section to do so. I want to get a new cert re-issued with a country code of 2 characters, but it's not allowing me to. It should have errored out before importing the certificate in my opinion.

I tried entering the PIV section on a Windows machine as well, but the same issue is happening.

---
Update:
I'm trying to remove the certificate via the piv tool. I authenticate successfully, but unable to delete:
```
$ yubico-piv-tool -averify-pin -P(pin)
Successfully verified PIN.

$ yubico-piv-tool -adelete-certificate -s 9a
Failed authentication with the application
```

Was able to delete the certificate using ykman though:
```
$ ykman piv delete-certificate 9a
Enter PIN: (pin)
```

**Comment by dagheyman** [comment]
Thanks for the report, this sounds like a bug indeed. Do you have an example certificate that triggers it? What did you use as value for the country code?

Good that you was able to delete the certificate with ykman, it would have been my suggested workaround.

**Comment by precurse** [comment]
`C=Another Country` was what I had it set to.

I can get the cert, but will need to wait until next week. It should be reproduceable with that, but let me know if you still need it.

Thanks!

**Comment by dagheyman** [comment]
What tool did you use to generate the certificate? I'm curious because OpenSSL doesn't let me specify long (invalid) country codes like that.
```
$ openssl req -new -newkey rsa:1024 -days 365 -nodes -x509 -keyout test.pem -out test.pem -subj "/C=Another\ Country/CN=example.com"
Generating a 1024 bit RSA private key
.............++++++
...................................................................++++++
writing new private key to 'test.pem'
-----
problems making Certificate Request
140162828345792:error:0D07A097:asn1 encoding routines:ASN1_mbstring_ncopy:string too long:../crypto/asn1/a_mbstr.c:109:maxsize=2
```

**Comment by dagheyman** [comment]
Seems like cryptography allows loading these invalid certs, but raises an error when reading out the issuer. See https://github.com/pyca/cryptography/issues/3857.

**Comment by precurse** [comment]
I use the cfssl (CloudFlare Go tool) to generate them. My signing (issuer) cert has the long country name set which, which I think is what it's choking on.

**Comment by dagheyman** [comment]
Thanks, good to know.

We managed to reproduce the issue with a cert with too long country code, so there are some fixes ready now. Expect this to be fixed in future versions of the tool.

Cert used for testing: 
```
-----BEGIN CERTIFICATE-----
MIICGDCCAYGgAwIBAgIJAOkI6jmkA4SrMA0GCSqGSIb3DQEBCwUAMCQxDDAKBgNVBAYTA2FiYzE
UMBIGA1UEAwwLZXhhbXBsZS5jb20wHhcNMTkwMjI3MTIyMzUyWhcNMjAwMjI3MTIyMzUyWjAjMQ
swCQYDVQQGEwJhYTEUMBIGA1UEAwwLZXhhbXBsZS5jb20wgZ8wDQYJKoZIhvcNAQEBBQADgY0AM
IGJAoGBAM3+iS9s7nLtmauZjV6/+JBxMX0wD7sR+symH7r6Yzpih7RfrHsaINSs6tnP+ikCH9mz
XQGmwBDrtV9zlwaMozMJN3jvIKK2I1rPpPTO4M3XXxYX8RohwpAE/wFEAY4iqUatAxRy7+mdhQQ
i/NgwvWnkmrx0jzqts6j2AQR09fiBAgMBAAGjUzBRMB0GA1UdDgQWBBSgYY+hDk7hgpckjM6nwR
fptV2paDAfBgNVHSMEGDAWgBSgYY+hDk7hgpckjM6nwRfptV2paDAPBgNVHRMBAf8EBTADAQH/M
A0GCSqGSIb3DQEBCwUAA4GBAD7mnzSxGl1q3yQeAHAxjCIYj/BdA5iC53G4Fe5zcYCnC0ZtFWwF
G1FPLYWe79OH2cnrq9ArkC1L2lxBoql+yfK8C7M+VdzGnna/HTj7jAYlzHSUyZ5pbmuwkpH6N16
oiUHIwzT634kQpjum3MfXzwtDQ5CxFtotJkWfIBMWLunI
-----END CERTIFICATE-----
```

**Comment by precurse** [comment]
Awesome! Glad you were able to reproduce it. Apologies I couldn't provide the exact cert, since I was leaving on vacation the day after I made the ticket :)

**Comment by ianarsenault** [comment]
I am using the library as a way to send alerts to expiring certificates. We have a certificate on file that is also throwing `ValueError: Country name must be a 2 character country code`. The issue seems to be this one particular certificate doesn't have a value for Country, along with a few other attributes.

```
Traceback (most recent call last):
  File "expring-cert.py", line 41, in <module>
    for attribute in cert.subject:
  File "/Users/ian/anaconda3/lib/python3.6/site-packages/cryptography/hazmat/backends/openssl/x509.py", line 111, in subject
    return _decode_x509_name(self._backend, subject)
  File "/Users/ian/anaconda3/lib/python3.6/site-packages/cryptography/hazmat/backends/openssl/decode_asn1.py", line 69, in _decode_x509_name
    attribute = _decode_x509_name_entry(backend, entry)
  File "/Users/ian/anaconda3/lib/python3.6/site-packages/cryptography/hazmat/backends/openssl/decode_asn1.py", line 60, in _decode_x509_name_entry
    return x509.NameAttribute(x509.ObjectIdentifier(oid), value, type)
  File "/Users/ian/anaconda3/lib/python3.6/site-packages/cryptography/x509/name.py", line 58, in __init__
    "Country name must be a 2 character country code"
ValueError: Country name must be a 2 character country code
```

Taking a look at the certificate which is breaking shows empty attributes for
- Organization
- Organization Unit
- Locality
- State
- Country

**Comment by dagheyman** [comment]
Should be fixed in the 1.1.1 release.
